presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -219,6 +239,214 @@ public void testDecimalExceedingPrecisionMax()
+                 "1234567890123456789012345678901234567890.1234567890",
+                "-1234567890123456789012345678901234567890.1234567890");
+        int largestScale = testData.stream().map(e -> e.length() - max(e.indexOf("."), 0) - 1).max(Integer::compareTo).orElse(0);
+        int largestIntegerSize = testData.stream().map(e -> (e + ".").replaceFirst("[-]", "").indexOf(".")).max(Integer::compareTo).orElse(0);
+        verify(largestIntegerSize > Decimals.MAX_PRECISION, format("largestIntegerSize must be greater than %d", Decimals.MAX_PRECISION));
+
+        try (TestTable testTable = new TestTable(
+                jdbcSqlExecutor,
+                "tpch.test_exceeding_max_decimal",
+                format("(d_col decimal(%d,%d))", typePrecision, typeScale),
+                testData)) {
+            String testTableName = testTable.getName().split("[.]")[1];
+            for (int scale : asList(0, 16, max(largestScale - 1, 0), largestScale, 38)) {
+                assertQuery(
+                        sessionWithDecimalMappingAllowOverflow(UNNECESSARY, scale),
+                        format("SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name = '%s'", testTableName),
+                        format("VALUES ('d_col', 'decimal(%d,%d)')",
+                                Decimals.MAX_PRECISION, calculateDecimalScaleMapping(typePrecision, typeScale, scale)));
+                if (scale < largestScale - 1) {
+                    assertQueryFails(
+                            sessionWithDecimalMappingAllowOverflow(UNNECESSARY, scale),
+                            format("SELECT d_col FROM %s", testTable.getName()),
+                            "Rounding necessary");
+                }
+                else {
+                    assertQueryFails(
+                            sessionWithDecimalMappingAllowOverflow(UNNECESSARY, scale),
+                            format("SELECT d_col FROM %s", testTable.getName()),
+                            "Decimal overflow");
+                }
+                assertQueryFails(
+                        sessionWithDecimalMappingAllowOverflow(HALF_UP, scale),
+                        format("SELECT d_col FROM %s", testTable.getName()),
+                        "Decimal overflow");
+            }
+            assertQuery(
+                    sessionWithDecimalMappingStrict(CONVERT_TO_VARCHAR),
+                    format("SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name = '%s'", testTableName),
+                    "VALUES ('d_col', 'varchar')");
+            assertQuery(
+                    sessionWithDecimalMappingStrict(CONVERT_TO_VARCHAR),
+                    format("SELECT d_col FROM %s", testTable.getName()),
+                    "VALUES " + testData.stream()
+                            .map(s -> format("('%s')", StringUtils.rightPad(s, s.split("[.]")[0].length() + typeScale + 1, "0")))
```

Comment:
I would just inline the expected values here. It will be much more easier for reader to understand what is the actual expected value.  Please do the same for every other test assertion.